src/main/java/io/lakefs/iceberg/LakeFSReporter.java

```diff
@@ -4,58 +4,61 @@
-import java.net.http.HttpRequest;
-import java.net.http.HttpResponse;
 import java.util.Base64;
 import java.util.HashMap;
 import java.util.Map;
 
 public class LakeFSReporter {
     private URI lakeFSEndpoint;
     private String authHeader;
-    private HttpClient httpClient;
+    private CloseableHttpAsyncClient httpClient;
     private final String reportClient = "iceberg-catalog/" + getClass().getPackage().getImplementationVersion();
 
-    private LakeFSReporter(){}
+    private LakeFSReporter() {
+    }
+
     public LakeFSReporter(Configuration hadoopConfig) {
         String lakeFSServerURL = hadoopConfig.get("fs.s3a.endpoint");
         this.lakeFSEndpoint = URI.create(StringUtils.stripEnd(lakeFSServerURL, "/") + "/api/v1/statistics");
         this.authHeader = generateBasicAuthHeader(hadoopConfig);
-        this.httpClient = HttpClient.newHttpClient();
+        this.httpClient = HttpAsyncClients.createDefault();
     }
 
     public void logOp(String op) {
         try {
-            Map<String, Object> reportMap = Map.of("class", "integration", "name", op, "count", 1);
-            HttpRequest request = generateRequest(reportMap);
-            httpClient.sendAsync(request, HttpResponse.BodyHandlers.ofString());
-        } catch (JsonProcessingException ignored) { }
+            Map<String, Object> reportMap = new HashMap<>();
+            reportMap.put("class", "integration");
+            reportMap.put("name", op);
+            reportMap.put("count", 1);
+            SimpleHttpRequest request = generateRequest(reportMap);
+            httpClient.execute(request, null);
+        } catch (JsonProcessingException ignored) {
+        }
     }
 
-    private HttpRequest generateRequest(Map<String, Object> body) throws JsonProcessingException {
+    private SimpleHttpRequest generateRequest(Map<String, Object> body) throws JsonProcessingException {
         String requestBody = prepareRequestBody(body);
-        return HttpRequest
-                .newBuilder()
-                .uri(this.lakeFSEndpoint)
-                .POST(HttpRequest.BodyPublishers.ofString(requestBody))
-                .header("Accept", "application/json")
-                .header("Content-Type", "application/json")
-                .header("Authorization", authHeader)
-                .header("X-Lakefs-Client", reportClient)
+        return SimpleRequestBuilder.post(this.lakeFSEndpoint).setBody(requestBody, ContentType.APPLICATION_JSON)
+                .addHeader("Accept", "application/json")
+                .addHeader("Authorization", authHeader)
+                .addHeader("X-Lakefs-Client", reportClient)
                 .build();
     }
 
     private String prepareRequestBody(Map<String, Object> requestMap) throws JsonProcessingException {
-        var statisticsRequest = new HashMap<String, Map<String, Object>[]>() {
+        Map<String, Map<String, Object>[]> statisticsRequest = new HashMap<String, Map<String, Object>[]>() {
```

Comment:
This initialization(?) syntax terrifies me.  In particular I don't understand why we need _two_ sets of curly brackets.

Also can we avoid repeating the type?
```suggestion
        Map<String, Map<String, Object>[]> statisticsRequest = new HashMap<>() {
```
or, if the reification type unification logic is too weak, perhaps
```suggestion
        Map<String, Map<String, Object>[]> statisticsRequest = new HashMap<String, Map<>[]>() {
```
?

Reply:
But why do we even need this weird syntax for a local object?  Just add the element on the next line, or use `ImmutableMap.of()`